devices/ubisys/h1_v1.7.json

```diff
@@ -0,0 +1,322 @@
+          },
+          "parse": {
+            "at": "0x0003",
+            "cl": "0x0201",
+            "ep": 1,
+            "eval": "Item.val = Attr.val * 100;",
+            "fn": "zcl:attr",
+            "mf": "0x10f2"
+          },
+          "default": 0
+        },
+        {
+          "name": "config/on"
+        },
+        {
+          "name": "config/reachable"
+        },
+        {
+          "name": "config/schedule",
+          "refresh.interval": 3660,
+          "read": {
+            "cl": "0x0201",
+            "cmd": "0x02",
+            "ep": "0x01",
+            "eval": "'7F01'",
+            "fn": "zcl:cmd"
+          }
+        },
+        {
+          "name": "config/schedule_on",
+          "static": true
+        },
+        {
+          "name": "config/unoccupiedheatsetpoint",
+          "refresh.interval": 3660,
+          "default": 1600
+        },
+        {
+          "name": "config/windowopendetectionenabled",
+          "read": {
+            "at": "0x0015",
+            "cl": "0x0201",
+            "ep": 1,
+            "mf": "0x10f2"
+          },
+          "write": {
+            "at": "0x0015",
+            "cl": "0x0201",
+            "dt": "0x10",
+            "ep": 1,
+            "eval": "Item.val",
+            "mf": "0x10f2"
+          },
+          "parse": {
+            "at": "0x0015",
+            "cl": "0x0201",
+            "ep": 1,
+            "eval": "Item.val = Attr.val;",
+            "mf": "0x10f2"
+          },
+          "default": false
+        },
+        {
+          "name": "state/lastupdated"
+        },
+        {
+          "name": "state/on",
+          "refresh.interval": 5,
+          "read": {
+            "fn": "none"
+          },
+          "parse": {
+            "at": "0x0008",
+            "cl": "0x0201",
+            "ep": 1,
+            "eval": "Item.val = Attr.val > 3;",
+            "fn": "zcl:attr"
+          }
+        },
+        {
+          "name": "state/temperature",
+          "refresh.interval": 3660,
+          "parse": {
+            "at": "0x0000",
+            "cl": "0x0201",
+            "ep": 1,
+            "eval": "Item.val = Attr.val;",
+            "fn": "zcl:attr"
+          },
+          "default": 0
+        },
+        {
+          "name": "state/utc",
+          "refresh.interval": 3660
+        },
+        {
+          "name": "state/valve",
+          "refresh.interval": 3660,
+          "default": 0
+        },
+        {
+          "name": "state/windowopen",
+          "read": {
+            "at": "0x0016",
+            "cl": "0x0201",
+            "ep": 1,
+            "mf": "0x10f2"
+          },
+          "write": {
+            "fn": "None"
+          },
+          "parse": {
+            "at": "0x0016",
+            "cl": "0x0201",
+            "ep": 1,
+            "eval": "if (Attr.val == 0) { Item.val = 'closed' } else if (Attr.val == 1) { Item.val = 'open' };",
+            "mf": "0x10f2"
+          },
```

Comment:
Please add the `fn` parameter and remove the write function.

Reply:
Strange. I added all of this via DDF Editor. Why is the `fn` parameter missing?